tests/src/ViewModel/SocialMediaSharersTest.php

```diff
@@ -0,0 +1,145 @@
+        $this->assertEquals('https://facebook.com/sharer/sharer.php?u='.urlencode($url), $facebookUrl);
+    }
+
+    /**
+     * @test
+     */
+    public function its_redditUrl_is_correctly_formed()
+    {
+        $url = 'https://elifesciences.org/articles/34901';
+        $redditUrl = (new SocialMediaSharers('Retrotransposons: On the move', $url))['redditUrl'];
+        $this->assertEquals('https://reddit.com/submit/?url='.urlencode($url), $redditUrl);
+    }
+
+    /**
+     * @test
+     */
+    public function its_emailUrl_is_correctly_formed()
+    {
+        $title = 'Retrotransposons: On the move';
+        $url = 'https://elifesciences.org/articles/34901';
+        $emailUrl = (new SocialMediaSharers($title, $url))['emailUrl'];
+        $this->assertEquals('mailto:?subject='.urlencode($title).'&amp;body='.urlencode($url), $emailUrl);
+    }
+
+    /**
+     * @test
+     */
+    public function its_twitterUrl_is_correctly_formed()
+    {
+        // One character shorter than theoretical max to account for space injected by Twitter between the text and url
+        $overallMaxLength = 139;
+
+        $tweets = [
+            [
+                'title' => 'Retrotransposons: On the move',
+                'url' => 'https://elifesciences.org/articles/34901',
+            ],
+            [
+                'title' => 'Early-Career Advisory Group: eLife welcomes 150 Ambassadors of good practice in science',
+                'url' => 'https://elifesciences.org/inside-elife/912b0679/early-career-advisory-group-elife-welcomes-150-ambassadors-of-good-practice-in-science',
+            ],
+        ];
+
+        foreach ($tweets as $tweet) {
+            $title = $tweet['title'];
+            $url = $tweet['url'];
+
+            $twitterUrl = (new SocialMediaSharers($title, $url))['twitterUrl'];
+            $this->assertStringStartsWith('https://twitter.com/intent/tweet/', $twitterUrl);
+
+            preg_match('/url=([^&]+)/', $twitterUrl, $urlMatch);
+            $this->assertEquals(urlencode($url), $urlMatch[1]);
+            $decodedUrl = urldecode($urlMatch[1]);
+
+            preg_match('/text=([^&]+)/', $twitterUrl, $textMatch);
+            $decodedText = urldecode($textMatch[1]);
+            if (urlencode($title) === $textMatch[1]) {
+                $this->assertLessThanOrEqual($overallMaxLength, strlen($decodedText.$decodedUrl));
+            } else {
+                // The title has been truncated to stop the overall tweet being too long
+                $ellipsis = ' &#8230;';
+                // -1 to account for the display of the ellipsis
+                $charCountTruncatedTitle = $overallMaxLength - strlen($decodedUrl) - 1;
+                $this->assertEquals(substr($title, 0, $charCountTruncatedTitle).$ellipsis, $decodedText);
+                $this->assertLessThanOrEqual($overallMaxLength + strlen($ellipsis) - 1, strlen($decodedText));
+            }
```

Comment:
Going to simplify these as they suffer from http://jasonrudolph.com/blog/2008/07/30/testing-anti-patterns-the-ugly-mirror/